packages/studio/src/components/ErrorComponentPreview.tsx

```diff
@@ -31,13 +31,14 @@ export default function ErrorComponentPreview(props: {
     setTooltipOpen(false);
   }, []);
 
-  document
-    .querySelector("iframe")
-    ?.contentDocument?.addEventListener("scroll", () => setTooltipOpen(false));
-
-  document.addEventListener("scroll", () => setTooltipOpen(false));
-
   useEffect(() => {
+    document
+      .querySelector("iframe")
+      ?.contentDocument?.addEventListener("scroll", () =>
+        setTooltipOpen(false)
+      );
+    document.addEventListener("scroll", () => setTooltipOpen(false));
```

Comment:
With the dependency array of this `useEffect`, new listeners would still be added any time one of those dependencies (`tooltipOpen`, `anchorId`, etc.) changed. I think these listeners should be in their own `useEffect` with an empty dependency array so they're only added on initial render, unless there's a reason they need to be updated after that. To remove a listener on unmount, you'll need to return a callback that removes the listener (see an example in `useWindowListener.js` in the React [docs](https://react.dev/reference/react/useEffect#custom-usewindowlistener-hook))

Reply:
Oh I see

Reply:
I think you still need to add the removal logic I mentioned?